### ISSUE TITLE
[Bug]: Flush 300s timeout when enablelevel zero segment and  concurrent insert_delete_flush 

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#29092]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.3.3 -> master-20231127-9c46788d-amd64
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.3.3
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy v2.3.3 milvus
2. create collection -> index -> insert 10m-128d -> flush -> index again -> load collection
3. concurrent tasks: 
   - search
   - scene_insert_delete_flush: insert -> delete -> flush
4. upgrade milvus to  master-20231127-9c46788d-amd64 during concurrent requests
5. flush 300s timeout
```
[2023-12-07 17:41:18,550 - ERROR - fouram]: RPC error: [flush], <MilvusException: (code=1, message=Retry timeout: 300s, message=wait for flush timeout, collection: fouram_CzTfnVoh)>, <Time:{'RPC start': '2023-12-07 17:34:45.383095', 'RPC error': '2023-12-07 17:41:18.550507'}> (decorators.py:128)
```

### Expected Behavior

flush succ

### Steps To Reproduce

```markdown
- argo: https://argo-workflows.zilliz.cc/archived-workflows/qa/ec1a8a83-e495-4417-8920-f9fbd419d1fb?nodeId=fouramf-rolling-upgrade-all-insert-1
```

### Milvus Log

pods:
```
delta-rolling-cluster-milvus-datanode-6ffdd7b685-6pn52            1/1     Running                           2 (3h51m ago)     17h     10.104.13.221   4am-node16   <none>           <none>
delta-rolling-cluster-milvus-datanode-6ffdd7b685-v44kx            1/1     Running                           0                 17h     10.104.4.165    4am-node11   <none>           <none>
delta-rolling-cluster-milvus-indexnode-678c98d8f7-bsk2d           1/1     Running                           0                 17h     10.104.5.72     4am-node12   <none>           <none>
delta-rolling-cluster-milvus-mixcoord-6dbcb5cd64-z7dg7            1/1     Running                           0                 17h     10.104.34.69    4am-node37   <none>           <none>
delta-rolling-cluster-milvus-proxy-694b48c9dd-8dq2j               1/1     Running                           0                 17h     10.104.33.149   4am-node36   <none>           <none>
delta-rolling-cluster-milvus-querynode-5bc975fbf7-mvcfp           1/1     Running                           0                 17h     10.104.6.9      4am-node13   <none>           <none>
```

**Comment by ThreadDao** [comment]
fixed

**Comment by ThreadDao** [comment]
@XuanYang-cn 
- Flush timeout 600s after upgrading from v2.3.3 -> master-20231221-a6eb7e5f
- argo: https://argo-workflows.zilliz.cc/archived-workflows/qa/0559b076-12ae-477f-bf48-debe26d4fb80?nodeId=fouramf-rolling-upgrade-all-cluster-1
- test steps:
 - deploy milvus with v2.3.3
 - during insert + delete + flush + search concurrently, upgrade milvus image
 - after upgrading, many flush timeout

- grafana: https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=prometheus&var-cluster=&var-namespace=qa-milvus&var-instance=upgrade-level-zero-2&var-collection=All&var-app_name=milvus&from=1703170800000&to=1703200019000

**Comment by ThreadDao** [comment]
fixed `master-20231225-13aa174b`
